File: app/lib/encryption/signup_result.dart
// signup function will return a SignUpResult
import 'dart:typed_data';

class SignUpResult {
  final Uint8List publicKey;
  final Uint8List encryptedPrivateKey;
  final Uint8List randomSaltOne;
  final Uint8List randomSaltTwo;

  SignUpResult({
    required this.publicKey,
    required this.encryptedPrivateKey,
    required this.randomSaltOne,
    required this.randomSaltTwo,
  });
}

File: backend/internal/types/sign_up_result.go
package types

type SignUpResult struct {
	PublicKey           []byte `json:"publicKey"`
	EncryptedPrivateKey []byte `json:"encryptedPrivateKey"`
	RandomSaltOne       []byte `json:"randomSaltOne"`
	RandomSaltTwo       []byte `json:"randomSaltTwo"`
	VerificationCode    []byte `json:"verificationCode"`
}
